docs/spec/README.md

```diff
@@ -1,3 +1,110 @@
 ---
 title: Specification
 ---
+
+This document defines the exact instructions required to be "App Config Compliant".
+What does that mean? App Config intends to be more than a library confined to the Node.js ecosystem.
+So in that spirit, we're providing instructions for language bindings to follow in order to interoperate.
+
+Does that mean we want N+1 ports of the App Config package? No, not really.
+This spec defines **minimal required support**, and App Config itself will be free to provide more features.
+
+## High Level Overview
+
+We'll separate parts of App Config:
+
+1. Loading App Config through files
+2. Loading App Config through an environment variable
+3. JSON Schema Validation
+4. Parsing extensions and environment-specific features
+
+So to be clear, **this spec will not define #1 or #4**. These are non-trivial
+parts of App Config, and would be a lot to replicate truthfully in every language.
+
+Instead, we'll try to outline #1 and #3. These may not describe _every_ feature
+that the App Config library supports, but App Config should faithfully fulfil the requirements.
+
+## Loading App Config through an environment variable
+
+App Config is defined as an environment variable. This variable is named `APP_CONFIG`.
+Libraries can support different names, but all should use `APP_CONFIG` by default.
+
+This environment variable should be a string, containing text that can be parsed as JSON.
+This JSON value should an object. Libraries should reject invalid JSON strings or non-objects, if possible.
+
+## JSON Schema Validation
+
+Whether the value contained in `APP_CONFIG` is valid is up to the schema.
+
+Schemas are JSON files, with the exact name `.app-config.schema.json`.
+This file should be present in the working directory, but libraries should provide an option for this.
+
+A missing schema file should result in an error, unless the library provides a non-validation opt-out option.
+
+Schemas should follow the exact specification of [JSON Schema](https://json-schema.org/specification.html).
+Users tend to use draft releases of this spec, so libraries are encouraged to support as many versions as possible.
+The exact version to use when a schema is ambiguous is up to library authors.
+
+Non-standard JSON Schema keywords should be ignored. In particular, the "secret" key is often used.
+Because library authors are only required to load environment variables, we don't require checking "secretness".
+
+**To be liberal, a library can skip JSON Schema validation**.
+We encourage authors not to do this, but we intend for this specification to be very minimal.
+So if your language has no off the shelf libraries for JSON Schema, it may be unreasonable.
+In these cases, we strongly encourage strict object-structure validation, at the least.
+We don't want users to receive unexpected values from App Config, which is exactly what the library is meant to avoid.
```

Comment:
I'm not sure this is something we want to compromise on. I get this may be non-trivial for some languages, but I think it's either in or out out of the spec. Allowing implementations to skip this could devalue the intent of the spec. 

IMO it *should* be in the spec, as this is a primary feature of `app-config` that is missing in other configuration loaders. Without this, I'm not sure `app-config` is any different from other config loaders.

Reply:
Right, it's definitely on the edge of "what's reasonable for us to expect a language to have?" (a simple standard JSON schema library). I think there's some value still (at least in strictly typed languages), but I can agree with you - let's take it out.